devito/ir/support/basic.py

```diff
@@ -426,7 +426,20 @@ def distance(self, other):
             else:
                 v = i - j
                 if v.is_Number and v.is_finite:
-                    return Vector(S.ImaginaryUnit)
+                    if i.is_Number and j.is_Number:
+                        return Vector(S.ImaginaryUnit)
+                    else:
+                        # For example:
+                        # self=W<u,[0,y]> and other=R<u,[0,y+1]>
+                        ret.append(v)
+
+                # Writing (reading) over an entire dimension, reading (writing)
+                # from one point
+
```

Comment:
nitpicking, I wouldn't normally say it, but since this is your learning phase and since this might be the first of many PRs... :

* Drop blank line;
* Move the "For example ..." on the same line as "# from one point. For example:" (after the for example, it's ok to break line)